include/boost/random/detail/qrng_base.hpp

```diff
@@ -0,0 +1,254 @@
+/* boost random/detail/qrng_base.hpp header file
+ *
+ * Copyright Justinas Vygintas Daugmaudis 2010-2018
+ * Distributed under the Boost Software License, Version 1.0. (See
+ * accompanying file LICENSE_1_0.txt or copy at
+ * http://www.boost.org/LICENSE_1_0.txt)
+ */
+
+#ifndef BOOST_RANDOM_DETAIL_QRNG_BASE_HPP
+#define BOOST_RANDOM_DETAIL_QRNG_BASE_HPP
+
+#include <stdexcept>
+#include <vector>
+#include <limits>
+
+#include <istream>
+#include <ostream>
+#include <sstream>
+
+#include <boost/cstdint.hpp>
+#include <boost/random/detail/operators.hpp>
+
+#include <boost/throw_exception.hpp>
+
+//!\file
+//!Describes the quasi-random number generator base class template.
+
+namespace boost {
+namespace random {
+
+namespace detail {
+
+template <typename T>
+inline bool are_close(T x, T y, T eps) // |x - y| <= eps
```

Comment:
This function is unnecessary.  The only place you're using this is to compare integers.

Reply:
I dislike this orphan function that is used once, too. But how would you suggest to compactly write the conjunctions in operator== without losing code flow?